src/mbgl/style/expression/number_format.cpp

```diff
@@ -0,0 +1,276 @@
+#include <mbgl/style/expression/number_format.hpp>
+#include <mbgl/style/conversion_impl.hpp>
+#include <mbgl/style/expression/formatted.hpp>
+
+#include <boost/algorithm/string/replace.hpp>
+
+#include <locale>
+#include <sstream>
+#include <iostream>
+#include <iomanip>
+
+namespace mbgl {
+namespace style {
+namespace expression {
+
+NumberFormat::NumberFormat(std::unique_ptr<Expression> number_,
+                           optional<std::unique_ptr<Expression>> locale_,
+                           optional<std::unique_ptr<Expression>> currency_,
+                           optional<std::unique_ptr<Expression>> minFractionDigits_,
+                           optional<std::unique_ptr<Expression>> maxFractionDigits_)
+    : Expression(Kind::NumberFormat, type::String),
+      number(std::move(number_))
+{
+      if (locale_)
+            locale = std::unique_ptr<Expression>(std::move(*locale_));
+
+      if (currency_)
+            currency = std::unique_ptr<Expression>(std::move(*currency_));
+
+      if (minFractionDigits_)
+            minFractionDigits = std::unique_ptr<Expression>(std::move(*minFractionDigits_));
+
+      if (maxFractionDigits_)
+            maxFractionDigits = std::unique_ptr<Expression>(std::move(*maxFractionDigits_));
+}
+
+EvaluationResult NumberFormat::evaluate(const EvaluationContext& params) const {
+    auto numberResult = number->evaluate(params);
+    if (!numberResult) {
+        return numberResult.error();
+    }
+    double evaluatedNumber = numberResult->get<double>();
+    
+    optional<std::string> evaluatedLocale;
+    if (locale) {
+        auto localeResult = (*locale)->evaluate(params);
+        if (!localeResult) {
+            return localeResult.error();
+        }
+        evaluatedLocale = toString(*localeResult);
+        // Replace '-' with '_' in the locale string and append '.utf8' to it
+        boost::replace_first(*evaluatedLocale, "-", "_");
```

Comment:
Can this be changed to std::string::find and std::string:replace?

Reply:
Done